From 28121ed889fb9a293925e8451e4397bdd8cb22c6 Mon Sep 17 00:00:00 2001
From: Anil Dsouza <anil.dsouza@einfochips.com>
Date: Mon, 14 Feb 2022 10:58:34 +0530
Subject: [PATCH 4/4] media: i2c: ap1320: Add support for arx3a0 sensor in
 ap1302 driver

This patch adds support for ARX3A0 in AP1302. It supports 120 FPS
with 560 x 560 resolution for this sensor.

Change-Id: I76dfd230e1d2835d9f288448047763d87775919e
Signed-off-by: Anil Dsouza <anil.dsouza@einfochips.com>
---
 drivers/media/i2c/ap1302.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index f63f2b514..035499a70 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -544,6 +544,20 @@ static const struct ap1302_sensor_info ap1302_sensor_info[] = {
 			{ "vddio", 0 },
 			{ NULL, 0 },
 		},
+	}, {
+		.model = "onnn,arx3a0",
+		.name = "arx3a0",
+		.i2c_addr = 0x36,
+		.resolution = { 560, 560},
+		.fps = 120,
+		.format = MEDIA_BUS_FMT_RGB565_2X8_LE,
+		.supplies = (const struct ap1302_sensor_supply[]) {
+			{ "vddpll", 0 },
+			{ "vaa", 0 },
+			{ "vdd", 0 },
+			{ "vddio", 0 },
+			{ NULL, 0 },
+		},
 	}
 };
 
@@ -2249,7 +2263,7 @@ static int ap1302_request_firmware(struct ap1302_device *ap1302)
 		return -EINVAL;
 	}
 
-	dev_dbg(ap1302->dev, "Requesting firmware %s\n", name);
+	dev_info(ap1302->dev, "Requesting firmware %s\n", name);
 
 	ret = request_firmware(&ap1302->fw, name, ap1302->dev);
 	if (ret) {
-- 
2.17.1

